### PR TITLE
Refactor publish and validation and servability for all file types

### PR DESCRIPTION
**Pull request by jessopb**
This one will probably have breaking changes when deployed unless:
 - site/custom/scss _variables is manually updated, and 
 - cli/defaults/siteConfig is manually updated to /site/config/siteConfig 

Improvements made:
 - more granular file size limit settings
 - no more hardcoded type restrictions
 - chainquery correctly assigns extensions in an extensible way

Todo: conditionally don't try to find file dimensions.

### CODE REVIEW

**Comment by kauffj** [comment]
It may make sense to look at adding versioning and a (crude) migration framework. I added to agenda for later today.

File: server/chainquery/models/ClaimModel.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+  serving: { customFileExtensions },
+} = require('@config/siteConfig');
 
 const getterMethods = {
   generated_extension() {
-    return mime.extension(this.content_type) ? mime.extension(this.content_type) : 'jpg';
+    logger.info('trying to generate extension', this.content_type);
+    if (customFileExtensions.hasOwnProperty(this.content_type)) {
+      return customFileExtensions[this.content_type];
+    } else return mime.extension(this.content_type) ? mime.extension(this.content_type) : 'jpg';
```
Comment (skhameneh) [inline server/chainquery/models/ClaimModel.js]: Please use (for legibility):
```
} else {
   return mime.extension(this.content_type) ? mime.extension(this.content_type) : 'jpg';
}
```

File: server/controllers/api/claim/publish/validateFileForPublish.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    customByContentType = {},
+  },
+} = publishing;
+
+const SIZE_MB = 1000000;
+
+const validateFileForPublish = file => {
+  let contentType = file.type;
+  let mediaType = contentType ? contentType.substr(0, contentType.indexOf('/')) : '';
+  console.log(file);
```
Comment (skhameneh) [inline server/controllers/api/claim/publish/validateFileForPublish.js]: Remove?

File: server/controllers/api/claim/publish/validateFileForPublish.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  if (Object.keys(customByContentType).includes(contentType)) {
+    if (file.size > customByContentType[contentType]) {
+      throw new Error(
+        `Sorry, type ${contentType} is limited to ${customByContentType[contentType] / SIZE_MB} MB.`
+      );
+    }
+  } else {
+    switch (mediaType) {
```
Comment (skhameneh) [inline server/controllers/api/claim/publish/validateFileForPublish.js]: Can make an object of `{ [mediaType]: maxSize }`, which would map to:
```
let mediaTypeLimit = mediaTypeLimits[mediaType] || false;
if (mediaTypeLimit) {
  if (file.size > mediaTypeLimit) {
    throw new Error(`Sorry, type ${mediaType} is limited to ${mediaTypeLimit / SIZE_MB} MB.`);
  }
}
```

File: utils/validateFileForPublish.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  if (Object.keys(customByContentType).includes(contentType)) {
+    if (file.size > customByContentType[contentType]) {
+      throw new Error(
+        `Sorry, type ${contentType} is limited to ${customByContentType[contentType] / SIZE_MB} MB.`
+      );
+    }
+  } else {
+    switch (mediaType) {
```
Comment (skhameneh) [inline utils/validateFileForPublish.js]: Much of this is duplicated in the other file I commented.

File: server/controllers/api/claim/publish/validateFileForPublish.js (hunk 3 above)
Comment (jessopb) [reply server/controllers/api/claim/publish/validateFileForPublish.js]: Done and DRY.

**Pull request merged**